ppa/archive/management/commands/generate_textcorpus.py

```diff
@@ -2,199 +2,238 @@
+
+"""
 import os
+from datetime import datetime
 import json
+import csv
 from django.core.management.base import BaseCommand
 from parasolr.django import SolrQuerySet
-from collections import defaultdict, OrderedDict
-import logging
-from tqdm import tqdm
-from typing import Tuple
-from contextlib import contextmanager
-import logging
-
-class SolrCorpus:
-    """Custom class to generate a text corpus from Solr"""
-
-    # Class attributes that rarely, if ever, need to change
-    DOC_ID_FIELD = "group_id_s"  # Solr field name for document identifier (not source_id, which is same across excerpts)
-    PAGE_ORDER_FIELD = "order"  # Solr field name for page ordering
-    OUTPUT_DOC_FIELDS = dict(
-        page_num_orig = 'label',
-        page_num_digi = 'order',
-        page_text = 'content',
-    )
-    PAGE_ID_FIELD = 'page_id'
-    WORK_ID_FIELD = 'work_id'
-    PAGE_NUM_FIELD = 'page_num_orig'
-    PAGE_SORT_FIELD = 'page_num_digi'
-
-    def __init__(self, path, doc_limit=-1):
-        """
-        A class encapsulating a Solr Client specification which yields
-       metadata and page data for PPA documents.
-
-        :param path: A string to a path for the corpus output.
-        :param doc_limit: Max no. of documents to process. The default of -1
-            means we process ALL documents found.
-        """
-        # root path for corpus
-        self.path = path
-
-        # limit docs queried
-        self.doc_limit = doc_limit
-
-        # subsequent paths
-        self.path_texts = os.path.join(self.path,'texts')
-        self.path_metadata = os.path.join(self.path,'metadata.json')
-        
-        # query to get initial results
-        results = SolrQuerySet().facet(self.DOC_ID_FIELD, limit=self.doc_limit)
-        # store page counts and doc ids
-        self.page_counts = results.get_facets().facet_fields[self.DOC_ID_FIELD]
-        self.doc_ids = self.page_counts.keys()
-        self.doc_count = len(self.doc_ids)
-    
-    @staticmethod
-    def _get_id(doc_id:str) -> str:
-        """Method to make a file-safe version of a document ID"""
-        return doc_id.replace('/','|')
-
-    def _get_meta_pages(self, doc_id:str) -> Tuple[dict,list]:
-        """Get metadata (dictionary) and pages (list of dictionaries) for a given document"""
-        
-        # get file safe work_id
-        work_id = self._get_id(doc_id)
-
-        # query
-        result = (
-            SolrQuerySet()
-            .search(**{self.DOC_ID_FIELD: doc_id})
-            .order_by(self.PAGE_ORDER_FIELD)
+from progressbar import progressbar
+import orjsonl
+from functools import cached_property
+from collections import deque
+
+DEFAULT_BATCH_SIZE = 10000
+
+
+class Command(BaseCommand):
+    """
+    Custom manage command to generate a text corpus from text indexed in Solr.
+    """
+
+    # fields we want from pages, in solr syntax: newfieldname:oldfieldname
+    PAGE_FIELDLIST = [
```

Comment:
any reason you didn't use a dict? (that's what we use in the aliased queryset in parasolr)

Reply:
oh ok, didn't realize— I thought parasolr wanted a list.

Reply:
you're generating the correct syntax! parasolr uses the dict to generate that syntax and a few other things: https://github.com/Princeton-CDH/parasolr/blob/main/parasolr/query/aliased_queryset.py#L20-L28

aliased queryset isn't in the parasolr docs either, is it? 🤦‍♀️ 😞